### ISSUE TITLE
cannot resolve generated rule type of `eslint-plugin-jsx-a11y`

### ISSUE THREAD

**Issue by kazupon** [antfu/eslint-typegen#7]
## summary

when we generate `eslint-plugin-jsx-a11y` rules with using `pluginsToRulesDTS`, the type of these rules cannot resolve.

## details

As an example of `jsx-a11y/accessible-emoji`, 
in `eslint-typegen.d.ts` generated by reproduction, the output is as follows:

`RuleOptions`
```ts
  /**
   * Enforce emojis are wrapped in `<span>` and provide screenreader access.
   * @see https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/tree/HEAD/docs/rules/accessible-emoji.md
   * @deprecated
   */
  'jsx-a11y/accessible-emoji'?: Linter.RuleEntry<JsxA11yAccessibleEmoji>
```

Type Declarations:
```ts
// ----- jsx-a11y/accessible-emoji -----
type _JsxA11YAccessibleEmoji_JsxA11YAccessibleEmoji = []|[{
  [k: string]: unknown | undefined
}]
```

In RuleOptions, it's output as `JsxA11yAccessibleEmoji`, but in Declarations it is output as `_JsxA11yAccessibleEmoji_JsxA11yAccessibleEmoji` and the type. The type is not correctly linked.

## Debugging tips

I’ve found out the code to see where the problem was and found that the issue was around compile from JSON schema.
https://github.com/antfu/eslint-typegen/blob/55af10090e34314363e040c3a9fde02be29db439/src/core.ts#L214-L223

In the `eslint-plugin-jsx-a11y` case, `schema.$id` and `name` do not match, so `_${name}_${resolved}` is resolved.

In `jsx-a11y/accessible-emoji` case,  these values are the following values:

- `name`: `'JsxA11yAccessibleEmoji'`
- `schema.$id`: `'JsxA11YAccessibleEmoji'`

Why it's normalized though, I have debugged deeply, and that’s normalized at `toSafeString` in `json-schema-to-typescript-lite`.

https://github.com/antfu/json-schema-to-typescript-lite/blob/b9f90414c1d74259d6674305ee3feb3fd3793f50/src/utils.ts#L160-L181

I hope my debugging info will help you debug. :)

### Reproduction

https://github.com/kazupon/eslint-typegen-repro1

### System Info

```Shell
System:
    OS: macOS 13.6.7
    CPU: (10) arm64 Apple M1 Max
    Memory: 251.56 MB / 64.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 18.19.1 - ~/.nvm/versions/node/v18.19.1/bin/node
  Browsers:
    Chrome: 127.0.6533.119
    Edge: 127.0.2651.98
    Firefox: 118.0.1
    Safari: 16.6
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)
